crates/kitsune2/src/lib.rs

```diff
@@ -37,3 +36,215 @@ pub fn default_builder() -> Builder {
+            })
+        }
+    }
+
+    #[derive(Debug)]
+    struct HolochainSpaceHandler;
+    impl SpaceHandler for HolochainSpaceHandler {}
+
+    async fn make_kitsune_space(
+        signal_server_url: &str,
+        bootstrap_server_url: &str,
+    ) -> DynSpace {
+        let mut kitsune_builder =
+            default_builder().with_default_config().unwrap();
+        kitsune_builder
+            .config
+            .set_module_config(&CoreBootstrapModConfig {
+                core_bootstrap: CoreBootstrapConfig {
+                    server_url: bootstrap_server_url.to_owned(),
+                    backoff_max_ms: 1000,
+                    ..Default::default()
+                },
+            })
+            .unwrap();
+        kitsune_builder.op_store = MemOpStoreFactory::create();
+        kitsune_builder
+            .config
+            .set_module_config(&Tx5TransportModConfig {
+                tx5_transport: Tx5TransportConfig {
+                    server_url: signal_server_url.to_owned(),
+                    signal_allow_plain_text: true,
+                },
+            })
+            .unwrap();
+        kitsune_builder
+            .config
+            .set_module_config(&K2GossipModConfig {
+                k2_gossip: K2GossipConfig {
+                    initiate_interval_ms: 100,
+                    min_initiate_interval_ms: 75,
+                    round_timeout_ms: 10000,
+                    ..Default::default()
+                },
+            })
+            .unwrap();
+
+        let kitsune_handler = Arc::new(HolochainKitsuneHandler);
+        let kitsune = kitsune_builder
+            .build(kitsune_handler.clone())
+            .await
+            .unwrap();
+        kitsune.space(TEST_SPACE_ID).await.unwrap()
+    }
+
+    #[tokio::test]
+    async fn two_node_gossip() {
+        enable_tracing();
+        let signal_server = SbdServer::new(Arc::new(sbd_server::Config {
+            bind: vec!["127.0.0.1:0".to_string()],
+            ..Default::default()
+        }))
+        .await
+        .unwrap();
+        let signal_server_url =
+            format!("ws://{}", signal_server.bind_addrs()[0]);
+
+        let bootstrap_server = TestBootstrapSrv::new(false).await;
+        let bootstrap_server_url = bootstrap_server.addr().to_string();
+
+        // Create 2 Kitsune instances and 1 space each.
+        let space_1 =
+            make_kitsune_space(&signal_server_url, &bootstrap_server_url).await;
+        let space_2 =
+            make_kitsune_space(&signal_server_url, &bootstrap_server_url).await;
+
+        // Insert ops into both spaces' op stores.
+        let (ops_1, op_ids_1) = create_op_list(100);
+        space_1
+            .op_store()
+            .process_incoming_ops(ops_1.clone())
+            .await
+            .unwrap();
+        let (ops_2, op_ids_2) = create_op_list(100);
+        space_2
+            .op_store()
+            .process_incoming_ops(ops_2.clone())
```

Comment:
Process incoming ops will make the ops available for "what's new" but there will be no DHT diff happening unless gossip is informed about these ops through the space with "inform_ops_stored". That will work fine for this test because all the ops are new. Just wanted to mention that.

Reply:
Yeah, I had that here but noticed it didn't make a diff(erence).